src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/PersistableSlidingWindow.java

```diff
@@ -72,6 +72,10 @@ protected synchronized void load(Path path) throws IOException {
     if (!enablePersistence) {
       return;
     }
+    if (!Files.exists(path)) {
+      LOG.debug("{}#load({}) called but the file doesn't exist", this.getClass().getSimpleName(), path);
```

Comment:
As no file exists in this path, will this be more of a warning logger level?

Reply:
I moved the validation responsibility to the caller and deleted this conditional. The ITs still pass.